e2e_tests/tests/experiment/experiment.py

```diff
@@ -914,3 +914,20 @@ def s3_checkpoint_config_no_creds() -> Dict[str, str]:
 
 def root_user_home_bind_mount() -> Dict[str, str]:
     return {"host_path": "/tmp", "container_path": "/root"}
+
+
+def has_at_least_one_checkpoint(exp_id: int) -> bool:
+    trials = experiment_trials(exp_id)
+    # Check if the most recent trial has any checkpoints.
+    if len(trials) > 0 and len(workloads_with_checkpoint(trials[len(trials) - 1].workloads)) > 0:
+        return True
+    return False
+
+
+def wait_for_at_least_one_checkpoint(experiment_id: int, timeout: int = 20) -> None:
+    for _ in range(timeout):
+        if has_at_least_one_checkpoint(experiment_id):
```

Comment:
20 seconds is a pretty tight default timing

Reply:
I would guess that Bradley would say 30 secs is pretty tight default timing :-)

checkpoints can take  a number of seconds to write, and if you miss this timeout the test is likely to fail.
You'll just end up raising it in the future where this is called I wager. 

So  maybe 120 secs?